test/core/test/__snapshots__/jest-expect.test.ts.snap

```diff
@@ -706,6 +706,48 @@ exports[`error equality 13`] = `
 }
 `;
 
+exports[`snapshotError tests > should capture error for missing property in actual object 1`] = `
+{
+  "actual": "Object {
+  "x": "foo",
+}",
+  "diff": "- Expected
++ Received
+
+  {
+-   "x": "bar",
+-   "y": undefined,
++   "x": "foo",
+  }",
```

Comment:
I think the desired diff for `toEqual` is to drop `y: undefined` since its existance doesn't affect `toEqual` assertion.

```
{
-  "x": "bar",
+ "x": "foo", 
}
```

Reply:
I have added an additional `filterUndefined` function which should get the job done